File: Fintech_Java/src/Usuario.java
import java.time.LocalDateTime;

public class Usuario {
    private String nome;
    private String email;
    private String senha;
    private LocalDateTime dataCadastro;
    private Conta conta;
    private Meta meta;

    public Usuario(){}

    public void cadastrar(String nome, String email, String senha) {
        this.nome = nome;
        this.email = email;
        this.senha = senha;
        this.dataCadastro = LocalDateTime.now();
        this.conta = new ContaCorrente();
    }

    public void registrarMeta(Meta meta){
        this.meta = meta;
    }

    public Conta getConta() {
        return conta;
    }

    public Meta getMeta() {
        return meta;
    }

    @Override
    public String toString() {
        return "Usuario{" +
                "nome='" + nome + '\'' +
                ", email='" + email + '\'' +
                ", senha='" + senha + '\'' +
                ", dataCadastro=" + dataCadastro +
                ", conta=" + conta +
                ", meta=" + meta +
                '}';
    }
}

File: fintech-app/app/types/usuario.ts
export interface Categoria {
  id?: number;
  nomeCategoria: string;
  tipoCategoria: string;
}

export interface Conta {
  id?: number;
  agencia: number;
  numero: number;
  tipoConta: "Conta Corrente" | "Conta Poupança" | "Conta Salário";
  saldo: number;
  dataCriacao?: string;
}

export interface Meta {
  id?: number;
  nome: string;
  dataCriacao?: string;
  valorAlvo: number;
  valorAtual: number;
  status: "Ativa" | "Concluída" | "Cancelada";
  dataAlvo: string;
  categoria?: Categoria;
}

export interface Transacao {
  id?: number;
  descricao: string;
  dataCriacao?: string;
  valor: number;
  tipo: "Pendente" | "Concluída" | "Cancelada";
  conta?: Conta;
  categoria?: Categoria;
}

export interface Usuario {
  id?: number;
  nome: string;
  email: string;
  senha?: string; // Senha é opcional no retorno
  dataCadastro?: string;
  contas?: Conta[];
  metas?: Meta[];
}

export interface Login {
  email: string;
  senha?: string;
}
